src/main/scala/Rational.scala

```diff
@@ -0,0 +1,65 @@
+case class Rational(n: Int, d: Int) {
+  require(d != 0)
```

Comment:
I think it's nice to add in a message for assertions to give the user feedback: https://www.scala-lang.org/api/current/scala/Predef$.html#require(requirement:Boolean,message:=%3EAny):Unit